### PR TITLE
[docs][ci] Adding steps to notify when /examples change

### PR DESCRIPTION
**Pull request by ronny-mysten**
## Description 

See title

## Test plan 

👀 

---

## Release notes

Check each box that your changes affect. If none of the boxes relate to your changes, release notes aren't required.

For each box you select, include information after the relevant heading that describes the impact of your changes that a user might notice and any actions they must take to implement updates. 

- [ ] Protocol: 
- [ ] Nodes (Validators and Full nodes): 
- [ ] Indexer: 
- [ ] JSON-RPC: 
- [ ] GraphQL: 
- [ ] CLI: 
- [ ] Rust SDK:
- [ ] REST API:

### CODE REVIEW

File: .github/workflows/docs-ci.yml [hunk 1]
```diff
@@ -29,3 +29,22 @@ jobs:
       - name: Build
         run: pnpm build
         working-directory: ./docs/site
+      - name: Check for changes in the examples directory
+        id: check_changes
+        run: |
+          if git diff --name-only ${{ github.event.before }} ${{ github.sha }} | grep '^examples/'; then
```
Comment (ebmifa) [inline .github/workflows/docs-ci.yml:35]: Are you sure this will work in bash?

**Review by ebmifa — CHANGES_REQUESTED** [review]

File: .github/workflows/docs-ci.yml (hunk 1 above)
Comment (ronny-mysten) [reply .github/workflows/docs-ci.yml:35]: Of course I'm not :)

Comment (ronny-mysten) [inline .github/workflows/docs-ci.yml:35]:
```suggestion
          if git diff --name-only $GITHUB_EVENT_BEFORE $GITHUB_SHA | grep '^examples/'; then
```

**Review by ebmifa — APPROVED** [review]

File: .github/workflows/docs-ci.yml [hunk 2]
```diff
@@ -29,3 +29,22 @@ jobs:
...
+        if: env.examplechanges == 'true' && github.event.action == 'opened'
+        uses: slackapi/slack-github-action@v1.24.0
+        with:
+          payload: |
+            {
+              "channel": "#tech-docs",
+              "text": "An open PR modifies files in the `examples/` directory. Review it: <https://github.com/${{ github.repository }}/pull/${{ github.event.number }}>"
+            }
+        env:
+          SLACK_BOT_TOKEN: ${{ secrets.SLACK_BOT_TOKEN }}
```
Comment (ebmifa) [inline .github/workflows/docs-ci.yml:50]: Add a newline back here 🙏🏼

Comment (ronny-mysten) [inline .github/workflows/docs-ci.yml:50]:
```suggestion
          SLACK_BOT_TOKEN: ${{ secrets.SLACK_BOT_TOKEN }}

```

**Pull request merged**